src/Agent.Worker/RetryHelper.cs

```diff
@@ -48,6 +60,49 @@ public async Task<T> Retry<T>(Func<Task<T>> action, Func<int, int> timeDelayInte
 
         }
 
+
+        public async Task RetryStep(Func<Task> action, Func<int, int> timeDelayInterval)
+        {
+            int retryCounter = 0;
+            do
+            {
+                using (new SimpleTimer($"RetryHelper Method:{action.Method} ", Debug))
+                {
+                    try
+                    {
+                        if (retryCounter > 0)
+                        {
+                            ExecutionContext.ReInitializeForceCompleted();
+                        }
+
+                        Debug($"Invoking Method: {action.Method}. Attempt count: {retryCounter}");
+                        await action();
+
+                        if (ExecutionContext.Result != TaskResult.Failed || ExhaustedRetryCount(retryCounter))
+                        {
+                            return;
+                        }
+                        else
+                        {
+                            string exceptionMessage = $"Task result ${ExecutionContext.Result}";
+                            ExecutionContext.Result = null;
+                            Warning($"RetryHelper encountered task failure, will retry (attempt #: {retryCounter + 1}) out of {this.MaxRetries}");
+                        }
+                    }
+                    catch (Exception ex)
+                    {
+                        if (!ShouldRetryStepOnException(ex) || ExhaustedRetryCount(retryCounter))
+                        {
+                            throw;
+                        }
+                        Warning($"RetryHelper encountered exception, will retry (attempt #: {retryCounter} {ex.Message}) ");
```

Comment:
Could it make sense to use loc strings instead?

Reply:
Not sure if it make sense to add it currently... We have localized strings for service only - probably it would make sense to take a look at it further as a separate work item.